agent/api/task_unix.go

```diff
@@ -25,6 +38,70 @@ func getCanonicalPath(path string) string { return path }
 
 // CgroupEnabled checks whether cgroups need be enabled at the task level
 func (task *Task) CgroupEnabled() bool {
-	//stub for now
+	// TODO: check for more conditions
+	// CPU + Mem -> true
+	// CPU -> true
+	// Mem -> false
+	// !CPU || !Mem -> false
+	if !utils.ZeroOrNil(task.VCPULimit) {
+		return true
+	}
 	return false
 }
+
+// BuildCgroupRoot helps build the task cgroup prefix
+// Example: /ecs/task-id
+func (task *Task) BuildCgroupRoot() (string, error) {
+	taskID, err := task.GetID()
+	if err != nil {
+		return "", errors.Wrapf(err, "task build cgroup root: unable to get task-id")
+	}
+
+	cgroupRoot := strings.Join([]string{config.DefaultTaskCgroupPrefix, taskID}, sepForwardSlash)
+	return cgroupRoot, nil
+}
+
+// BuildLinuxResourceSpec returns a linuxResources object for the task cgroup
+func (task *Task) BuildLinuxResourceSpec() specs.LinuxResources {
+	linuxResourceSpec := specs.LinuxResources{}
+
+	if !(utils.ZeroOrNil(task.VCPULimit)) {
+		taskCPUPeriod := uint64(defaultCPUPeriod)
+		taskCPUQuota := int64(task.VCPULimit * defaultCPUPeriod)
+
+		linuxResourceSpec.CPU = &specs.LinuxCPU{
+			Quota:  &taskCPUQuota,
+			Period: &taskCPUPeriod,
+		}
+	}
+
+	if !(utils.ZeroOrNil(task.MemoryLimit)) {
+		taskMemoryLimit := int64(task.MemoryLimit)
+		linuxResourceSpec.Memory = &specs.LinuxMemory{
+			Limit: &taskMemoryLimit,
+		}
+	}
+
+	return linuxResourceSpec
+}
+
+// platformHostConfigOverride to override platform specific feature sets
+func (task *Task) platformHostConfigOverride(hostConfig *docker.HostConfig) {
+	// Override cgroup parent
+	task.overrideCgroupParent(hostConfig)
+}
+
+// overrideCgroupParent updates hostconfig with cgroup parent when task cgroups
+// are enabled
+func (task *Task) overrideCgroupParent(hostConfig *docker.HostConfig) {
+	if !task.CgroupEnabled() {
+		return
+	}
+
+	cgroupRoot, err := task.BuildCgroupRoot()
+	if err != nil {
```

Comment:
Why this error was ignored, shouldn't  the task moved to be stopped when it can't get the group root?

Reply:
This should fail only when the `taskARN` is invalid and I thought that to be highly infrequent. However, I can return an error to make it more semantically correct.